sklearn/tree/_criterion.pyx

```diff
@@ -962,6 +963,108 @@ cdef class MSE(RegressionCriterion):
         impurity_left[0] /= self.n_outputs
         impurity_right[0] /= self.n_outputs
 
+cdef class MAE(RegressionCriterion):
+    """Mean absolute error impurity criterion
+    """
+    cdef void node_value(self, double* dest) nogil:
+        """Computes the node value of samples[start:end] into dest."""
+        with gil:
+            print "entered node_value"
+        cdef double* sample_weight = self.sample_weight
+        cdef SIZE_t* samples = self.samples
+        
+        cdef DOUBLE_t* y = self.y
+        cdef SIZE_t start = self.start
```

Comment:
I noticed that other code in this file does something like this quite commonly, setting a local variable to the value of a class variable. Is the any reason not to use `self.start` directly here?